blockchain/chainio.go

```diff
@@ -1167,6 +1167,44 @@ func (b *BlockChain) initChainState() error {
 			return err
 		}
 
+		// Load all of the headers from the data for the known best
+		// chain and construct the block index accordingly.  Since the
+		// number of nodes are already known, perform a single alloc
+		// for them versus a whole bunch of little ones to reduce
+		// pressure on the GC.
```

Comment:
Thanks!  I'm not following what you mean since the index is using a map where the entries are a pointer to an existing block node.  One of the big things this PR aims to do is make it so there is only ever one single block node per block in memory.  That will allow them to be compared by pointer in other code I have in mind which is significantly faster than eating derefs + hash comparisons.